### PR TITLE
GBE-549: TMS controller extended to publish all TMS metadata + support

### PR DESCRIPTION
**Pull request by jandm**
TMS metadata support + common TMS profiles (web mercator, geodetic, local)

### CODE REVIEW

**Comment by oliverm-be** [comment]
Oliver May: +1 (binding)

**Comment by dosi-bingov** [comment]
My vote +1

On 01/27/2015 05:28 PM, Oliver May wrote:
>
> Oliver May: +1 (binding)
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/geomajas/geomajas-project-server/pull/76#issuecomment-71678009>.
>

-- 

Dosi Bingov / Java developer
skype:dosi.dfc/ dosi.bingov@dfc.be <mailto:dosi.bingov@dfc.be>

DFC Software Engineering Office: +32 9 236 61 96 / Fax: +32 9 236 54 12
Brugsesteenweg 587 9030 Mariakerke , Belgium
www.geomajas.org

File: plugin/rasterizing/rasterizing/src/main/java/org/geomajas/plugin/rasterizing/tms/TmsService.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.geomajas.global.GeomajasException;
+
+/**
+ * The TMS service is responsible for publishing the TMS metadata (vector layers only!). See
+ * {@link org.geomajas.plugin.rasterizing.mvc.TmsController} for the actual rendering of the tiles.
+ * 
+ * @author Jan De Moerloose
+ *
+ */
+public interface TmsService {
```
Comment (janvenstermans) [inline plugin/rasterizing/rasterizing/src/main/java/org/geomajas/plugin/rasterizing/tms/TmsService.java]: Should this better be called TmsWriterService ?

File: plugin/rasterizing/rasterizing/src/test/java/org/geomajas/plugin/rasterizing/tms/TmsServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+				"        <TileMap href=\"http://test/tms/1.0.0/layerBeansMultiPolygon@EPSG:4326/layerBeansMultiPolygonStyleInfo\" profile=\"global-geodetic\" srs=\"EPSG:4326\" title=\"layerBeansMultiPolygon\"/>\n" + 
+				"        <TileMap href=\"http://test/tms/1.0.0/layerBeansMultiPolygon@EPSG:4326/layerBeansMultiPolygonStyleInfo\" profile=\"local\" srs=\"EPSG:4326\" title=\"layerBeansMultiPolygon\"/>\n" + 
+				"        <TileMap href=\"http://test/tms/1.0.0/layerBeansPoint@EPSG:3857/layerBeansPointStyleInfo\" profile=\"global-mercator\" srs=\"EPSG:3857\" title=\"layerBeansPoint\"/>\n" + 
+				"        <TileMap href=\"http://test/tms/1.0.0/layerBeansPoint@EPSG:4326/layerBeansPointStyleInfo\" profile=\"global-geodetic\" srs=\"EPSG:4326\" title=\"layerBeansPoint\"/>\n" + 
+				"        <TileMap href=\"http://test/tms/1.0.0/layerBeansPoint@EPSG:4326/layerBeansPointStyleInfo\" profile=\"local\" srs=\"EPSG:4326\" title=\"layerBeansPoint\"/>\n" + 
+				"    </TileMaps>\n" + 
+				"</TileMapService>\n" + 
+				"", sw.toString());
+	}
```
Comment (janvenstermans) [inline plugin/rasterizing/rasterizing/src/test/java/org/geomajas/plugin/rasterizing/tms/TmsServiceTest.java]: Can expected xml code be set in a separate xml file, or using CDATA? DOn't know if it is possible, but it seems better than String on different lines.

File: plugin/rasterizing/rasterizing/src/main/java/org/geomajas/plugin/rasterizing/tms/TmsService.java (hunk 1 above)
Comment (jandm) [inline plugin/rasterizing/rasterizing/src/main/java/org/geomajas/plugin/rasterizing/tms/TmsService.java]: I consider it a TMS service (or metadata service) but it looks indeed like the focus is on the writing now. To be reevaluated when we turn it into API ?

**Pull request merged**